src/main/java/DukeExceptionHandler.java

```diff
@@ -0,0 +1,99 @@
+            } else if (replyArr[0].equals("deadline")) {
+                exception = new MissingDescriptionException(InputType.DEADLINE);
+            } else {
+                exception = new MissingDescriptionException(InputType.EVENT);
+            }
+        // if the userInput has the action keywords but no task number
+        } else if (replyArr.length == 1 && (replyArr[0].equals("done")
+                || replyArr[0].equals("delete"))) {
+            if (replyArr[0].equals("done")) {
+                exception = new MissingTaskIndexException(InputType.DONE);
+            } else {
+                exception = new MissingTaskIndexException(InputType.DELETE);
+            }
+        }
+        return exception;
+    }
+
+    public static String checkForException_new(String userInput) throws DukeException {
+        String[] replyArr = null;
+        DukeException exception = null;
+        if (userInput.length() > 0) {
+            replyArr = userInput.split(" ");
+        }
+        if (userInput.length() == 0) {
+            exception = new EmptyTaskException();
+            // shortest userInput allowed is "list"
+            // also unexpected if it doesn't contain/equal to any of the keywords:
+            // "todo, deadline, event, done, delete, list, find"
+        } else if (userInput.length() < 4
+                || (!userInput.contains("todo") && !userInput.contains("deadline")
+                && !userInput.contains("event") && !userInput.contains("done")
+                && !userInput.contains("delete") && !userInput.contains("list")
+                && !userInput.contains("find"))) {
+            exception = new UnexpectedInputException();
+            // if the userInput has the task keywords but not description
+            // future: need to check if deadline and event have "/by" and "/at"
+            // can try to use string.startsWith
+        } else if (replyArr.length == 1 && (replyArr[0].equals("todo")
+                || replyArr[0].equals("deadline")
+                || replyArr[0].equals("event"))) {
+            if (replyArr[0].equals("todo")) {
+                exception = new MissingDescriptionException(InputType.TODO);
+            } else if (replyArr[0].equals("deadline")) {
+                exception = new MissingDescriptionException(InputType.DEADLINE);
+            } else {
+                exception = new MissingDescriptionException(InputType.EVENT);
+            }
+            // if the userInput has the action keywords but no task number
+        } else if (replyArr.length == 1 && (replyArr[0].equals("done")
+                || replyArr[0].equals("delete"))) {
+            if (replyArr[0].equals("done")) {
+                exception = new MissingTaskIndexException(InputType.DONE);
+            } else {
+                exception = new MissingTaskIndexException(InputType.DELETE);
+            }
+        } else if (!replyArr[0].trim().equals("todo") && !replyArr[0].trim().equals("event")
```

Comment:
it might make sense here to factor out `replyArr[0].trim()` to a separate variable, so it's less repetitive?

also, you might want to use `string.strip()` instead of `trim()`, since the former is unicode compliant (:

Reply:
also, regex might be a good idea here as well?